Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryDeferredRequestAuditTransformsTest.java

```diff
@@ -0,0 +1,222 @@
+        final String localIp = "10.10.10.10";
+        final String remoteIp = "16.14.13.12";
+        final String remoteObjectUrl = "http://" + remoteIp + ":9090/source/AuditService";
+        final String wsRequestUrl = "http://" + remoteIp + ":9090/AuditService";
+
+        Properties webContextProperties = new Properties();
+        webContextProperties.setProperty(NhincConstants.WEB_SERVICE_REQUEST_URL, wsRequestUrl);
+        webContextProperties.setProperty(NhincConstants.REMOTE_HOST_ADDRESS, remoteIp);
+        webContextProperties.setProperty(NhincConstants.INBOUND_REPLY_TO,
+            NhincConstants.WSA_REPLY_TO);
+
+        PatientDiscoveryDeferredRequestAuditTransforms transforms = new PatientDiscoveryDeferredRequestAuditTransforms() {
+            @Override
+            protected String getLocalHostAddress() {
+                return localIp;
+            }
+
+            @Override
+            protected String getRemoteHostAddress(Properties webContextProperties) {
+                if (webContextProperties != null && !webContextProperties.isEmpty() && webContextProperties
+                    .getProperty(NhincConstants.REMOTE_HOST_ADDRESS) != null) {
+
+                    return webContextProperties.getProperty(NhincConstants.REMOTE_HOST_ADDRESS);
+                }
+                return AuditTransformsConstants.ACTIVE_PARTICIPANT_UNKNOWN_IP_ADDRESS;
+            }
+
+            @Override
+            protected String getWebServiceUrlFromRemoteObject(NhinTargetSystemType target, String serviceName) {
+                return remoteObjectUrl;
+            }
+        };
+
+        AssertionType assertion = createAssertion();
+        LogEventRequestType auditResponse = transforms.transformResponseToAuditMsg(
+            TestPatientDiscoveryMessageHelper.createPRPAIN201305UV02Request("Gallow", "Younger", "M", "01-12-2967",
+                "1.1", "D123401", "2.2", "abd3453dcd24wkkks545"), new MCCIIN000002UV01(), assertion, createNhinTarget(),
+            NhincConstants.AUDIT_LOG_INBOUND_DIRECTION, NhincConstants.AUDIT_LOG_ENTITY_INTERFACE, Boolean.FALSE,
+            webContextProperties, NhincConstants.PATIENT_DISCOVERY_DEFERRED_REQ_SERVICE_NAME);
+
+        testGetEventIdentificationType(auditResponse, NhincConstants.PATIENT_DISCOVERY_DEFERRED_REQ_SERVICE_NAME,
+            Boolean.FALSE);
+        testGetActiveParticipantSource(auditResponse, Boolean.FALSE, webContextProperties, remoteIp);
+        testGetActiveParticipantDestination(auditResponse, Boolean.FALSE, webContextProperties, wsRequestUrl);
+        assertNotNull("ParticipantObjectIdentification is null", auditResponse.getAuditMessage().
+            getParticipantObjectIdentification());
+        assertFalse("ParticipantObjectIdentification is empty", auditResponse.getAuditMessage().
+            getParticipantObjectIdentification().isEmpty());
+        assertEquals("ParticipantObjectIdentification size is 1", auditResponse.getAuditMessage().
```

Comment:
This message is wrong; I would suggest something along the lines of "ParticipantObjectIdentification should only have 1 entry" (at this point, we know that there is at least 1 entry because we already checked "isEmpty()"), but pick a message you're comfortable with.